src/main/kotlin/org/rust/lang/core/parser/RustParserUtil.kt

```diff
@@ -726,4 +726,9 @@ object RustParserUtil : GeneratedParserUtilBase() {
     fun parseCodeBlockLazy(builder: PsiBuilder, level: Int): Boolean {
         return PsiBuilderUtil.parseBlockLazy(builder, LBRACE, RBRACE, BLOCK) != null
     }
+
+    @JvmStatic
+    fun parserPatWithoutOrPats(builder: PsiBuilder): Boolean {
```

Comment:
`WithoutOrPats` suffix is quite confusing because it makes me think that it doesn't parse or_patterns at all. But it doesn't pars or_patterns only in top level. Maybe it's better not to introduce new function with a confusing name and just inline it?

Reply:
It cant' be inlined because `RustParser.SimplePat` is package-private because of weird grammar-kit visibility rules

Reply:
Well, what do you think about `parseSimplePat` name? `SimplePat` is already used in our grammar so it shouldn't confuse reader

Reply:
Ok